.github/workflows/qiita-ci.yml

```diff
@@ -0,0 +1,207 @@
+          redbiom admin load-sample-metadata-search --metadata `pwd`/qiita_db/support_files/test_data/templates/1_19700101-000000.txt
+          redbiom admin load-sample-data --table `pwd`/qiita_db/support_files/test_data/processed_data/1_study_1001_closed_reference_otu_table.biom --context qiita-test --tag 4
+          redbiom admin load-sample-data --table `pwd`/qiita_db/support_files/test_data/processed_data/1_study_1001_closed_reference_otu_table-for_redbiom_tests.biom --context qiita-test --tag 5
+
+          echo "4. Setting up nginx"
+          mkdir -p /usr/share/miniconda/envs/qiita/var/run/nginx/
+          nginx -c ${PWD}/qiita_pet/nginx_example.conf
+
+          echo "5. Setting up qiita"
+          conda activate qiita
+          qiita-env make --no-load-ontologies
+          qiita-test-install
+          qiita plugins update
+
+          echo "6. Starting supervisord => multiple qiita instances"
+          supervisord -c ${PWD}/qiita_pet/supervisor_example.conf
+          sleep 10
+          cat /tmp/supervisord.log
+
+          echo "7. Starting plugins"
+          conda deactivate
+          conda activate qtp-biom
+          export QIITA_CLIENT_DEBUG_LEVEL=DEBUG
+          start_biom https://localhost:8383 register ignored
+          conda deactivate
+
+          echo "8. Setting up SSH"
+          ssh-keygen -t rsa-sha2-256 -b 4096 -N '' -f $PWD/qiita_ware/test/test_data/test_key
+          mkdir ~/.ssh/
+          cp $PWD/qiita_ware/test/test_data/test_key* ~/.ssh/
+          cat ~/.ssh/test_key.pub > ~/.ssh/authorized_keys
+          chmod 600 $PWD/qiita_ware/test/test_data/test_key*
+          chmod 600 ~/.ssh/*
+          chmod 700 ~/.ssh/
+          scp -o StrictHostKeyChecking=no -i $PWD/qiita_ware/test/test_data/test_key $USER@localhost:$PWD/qiita_ware/test/test_data/random_key $PWD/qiita_ware/test/test_data/random_key_copy
+
+      - name: Main tests
+        shell: bash -l {0}
+        env:
+          COVER_PACKAGE: ${{ matrix.COVER_PACKAGE }}
+        run: |
+          conda activate qiita
+          bash -c 'source /home/runner/.profile; conda activate qtp-biom; start_biom --help'
+          export QIITA_SERVER_CERT=`pwd`/qiita_core/support_files/server.crt
+          export QIITA_CONFIG_FP=`pwd`/qiita_core/support_files/config_test.cfg
+          export REDBIOM_HOST="http://localhost:7379"
+          nosetests $COVER_PACKAGE --with-doctest --with-coverage --with-timer -v --cover-package=$COVER_PACKAGE -e 'test_submit_EBI_parse_EBI_reply_failure' -e 'test_full_submission'
+
+          # killing the qiita server to run the next commands
+          QIITA_PID=`cat /tmp/supervisord.pid`
+          kill $QIITA_PID
+          sleep 10
+          if [[ "$COVER_PACKAGE" == *"qiita_db"* ]]; then test_data_studies/commands.sh; all-qiita-cron-job; fi
+
+      - uses: codecov/codecov-action@v1
+        with:
+         file: codecov.yml
+         fail_ci_if_error: true
+
+  lint:
```

Comment:
Up to you but maybe the `main` job could depend on the `lint` job so that the full test suite isn't run before you find there's a syntax error or something like that. That would also mean that the tests aren't run if there are linting errors which might be undesirable.